nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-5-processors/src/main/java/org/apache/nifi/processors/elasticsearch/PutElasticsearch5.java

```diff
@@ -196,6 +205,7 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                 final String index = context.getProperty(INDEX).evaluateAttributeExpressions(file).getValue();
                 final String docType = context.getProperty(TYPE).evaluateAttributeExpressions(file).getValue();
                 final String indexOp = context.getProperty(INDEX_OP).evaluateAttributeExpressions(file).getValue();
+                final Long version = version_attribute != null ? Long.parseLong(file.getAttribute(version_attribute)) : null;
```

Comment:
FYI you could also call `file.getAttribute(version_attribute).asLong()`. Also if the field is supposed to contain a Long value (if set), then you should consider using a LONG_VALIDATOR for that field (rather than the NON_EMPTY_VALIDATOR I suggested in my other comment). This will avoid a possible NumberFormatException if someone enters invalid input.

Reply:
From my understanding this field is suppose to take the flowfile attribute name, thats why I went with  ATTRIBUTE_KEY_VALIDATOR .

I'm updating the code to use .asLong() instead ! :)

Reply:
For some reason can't use the .as Long(), file.getAttribute(version_attribute) returns a String per default. Suggestions ?

Reply:
My mistake, was thinking of PropertyDescriptor.asLong(), if you change from an attribute to using the property (with EL support, see above comments). asLong() doesn't apply here unless you change from requiring an attribute vs a property value, sorry about that.